dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/util/ContentletUtilTest.java

```diff
@@ -64,251 +70,138 @@ public static void prepare () throws Exception {
         language = APILocator.getLanguageAPI().getDefaultLanguage();
 
         categoryAPI = APILocator.getCategoryAPI();
-        contentletAPI = APILocator.getContentletAPI();
-        contentTypeAPI = APILocator.getContentTypeAPI( user, false );
-        fieldAPI = APILocator.getContentTypeFieldAPI();
 
-        defaultHost = APILocator.getHostAPI().findDefaultHost( user, false );
+        host = new SiteDataGen().nextPersisted();
     }
 
+    private static final String CATEGORY_NAMES = "categoryNames";
+    private static final String ALL_CATEGORIES_INFO = "allCategoriesInfo";
+
+    @DataProvider
+    public static Object[] categoryTestCases() {
+        return new String[] {
+                CATEGORY_NAMES, ALL_CATEGORIES_INFO
+        };
+    }
 
     /**
      * https://github.com/dotCMS/core/issues/11751
      */
     @Test
-    public void validateContentPrintableMapMethodReturnProperCategories() throws Exception{
+    @UseDataProvider("categoryTestCases")
+    public void validateContentPrintableMapMethodReturnProperCategories(
+            final String categoryInfoType) throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4885353)